pdns/iputils.cc

```diff
@@ -323,61 +327,67 @@ int sendOnNBSocket(int fd, const struct msghdr *msgh)
 void fillMSGHdr(struct msghdr* msgh, struct iovec* iov, cmsgbuf_aligned* cbuf, size_t cbufsize, char* data, size_t datalen, ComboAddress* addr)
 {
   iov->iov_base = data;
-  iov->iov_len  = datalen;
+  iov->iov_len = datalen;
 
   memset(msgh, 0, sizeof(struct msghdr));
 
   msgh->msg_control = cbuf;
   msgh->msg_controllen = cbufsize;
   msgh->msg_name = addr;
   msgh->msg_namelen = addr->getSocklen();
-  msgh->msg_iov  = iov;
+  msgh->msg_iov = iov;
   msgh->msg_iovlen = 1;
   msgh->msg_flags = 0;
 }
 
 // warning: various parts of PowerDNS assume 'truncate' will never throw
 void ComboAddress::truncate(unsigned int bits) noexcept
 {
-  uint8_t* start;
-  int len=4;
-  if(sin4.sin_family==AF_INET) {
-    if(bits >= 32)
+  uint8_t* start{};
+  int len = 4;
+  if (sin4.sin_family == AF_INET) {
+    if (bits >= 32) {
       return;
-    start = (uint8_t*)&sin4.sin_addr.s_addr;
-    len=4;
+    }
+    start = reinterpret_cast<uint8_t*>(&sin4.sin_addr.s_addr); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+    len = 4;
   }
   else {
-    if(bits >= 128)
+    if (bits >= 128) {
       return;
-    start = (uint8_t*)&sin6.sin6_addr.s6_addr;
-    len=16;
+    }
+    start = reinterpret_cast<uint8_t*>(&sin6.sin6_addr.s6_addr); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+    len = 16;
   }
 
-  auto tozero= len*8 - bits; // if set to 22, this will clear 1 byte, as it should
+  auto tozero = len * 8 - bits; // if set to 22, this will clear 1 byte, as it should
 
-  memset(start + len - tozero/8, 0, tozero/8); // blot out the whole bytes on the right
+  memset(start + len - tozero / 8, 0, tozero / 8); // blot out the whole bytes on the right NOLINT(cppcoreguidelines-pro-bounds-pointer-arithmetic)
 
-  auto bitsleft=tozero % 8; // 2 bits left to clear
+  auto bitsleft = tozero % 8; // 2 bits left to clear
 
   // a b c d, to truncate to 22 bits, we just zeroed 'd' and need to zero 2 bits from c
   // so and by '11111100', which is ~((1<<2)-1)  = ~3
-  uint8_t* place = start + len - 1 - tozero/8;
-  *place &= (~((1<<bitsleft)-1));
+  uint8_t* place = start + len - 1 - tozero / 8; // NOLINT(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+  *place &= (~((1 << bitsleft) - 1));
 }
 
-size_t sendMsgWithOptions(int fd, const char* buffer, size_t len, const ComboAddress* dest, const ComboAddress* local, unsigned int localItf, int flags)
+size_t sendMsgWithOptions(int fileDesc, const char* buffer, size_t len, const ComboAddress* dest, const ComboAddress* local, unsigned int localItf, int flags)
 {
-  struct msghdr msgh;
-  struct iovec iov;
+  struct msghdr msgh
+  {
+  };
+  struct iovec iov
+  {
+  };
```

Comment:
```suggestion
  msghdr msgh{};
  iovec iov{};
```

Reply:
clang-format insists, unless I remove `struct`.

Reply:
Yes, that's why I'm suggesting to remove `struct` :)

Reply:
Ah, did not spot that in your suggestion :-(